### PR TITLE
Fix reading of enhanced Structures nested inside Sequences, plus ToolsUI and NcdumpW tweaks

### PR DESCRIPTION
**Pull request by cwardgar**
Fixes [netCDFJava !PEH-614230](https://andy.unidata.ucar.edu/esupport/staff/index.php?_m=tickets&_a=viewticket&ticketid=29223).

### CODE REVIEW

**Comment by cwardgar** [comment]
Hold off on merging this. Don responded with another problem. I'll try to fix that first.

**Comment by cwardgar** [comment]
The Sequence/Structure issues were _really_ hard to debug. I guess nobody's really using BUFR, or else we'd have found them earlier.

In any event, this is ready to merge now. See additional commits for notes.

**Comment by lesserwhirls** [comment]
Well dang. Oh well, one can dream of a BUFR savior, yes?

Let's have @DennisHeimbigner take a look (given that he is a code owner) and we'll merge after that.

Thanks @cwardgar !

**Comment by cwardgar** [comment]
@DennisHeimbigner This was the PR I was referring to.

**Comment by lesserwhirls** [comment]
Totally forgot to bring this up at the meeting today - @DennisHeimbigner , do these changes look good to you?

**Comment by lesserwhirls** [comment]
We'll just go with it so that we can catch up with other PRs.

File: cdm/src/main/java/ucar/ma2/ArrayStructureMA.java [hunk 1]
```diff
@@ -185,18 +187,57 @@ public ArrayStructureMA(StructureMembers members, int[] shape, StructureData[] s
...
+    if (from.getSize() > 0) {
+      ArrayStructureMA to = new ArrayStructureMA(new StructureMembers(from.getStructureMembers()), from.getShape());
+      for (StructureMembers.Member m : from.getMembers()) {
+        to.setMemberArray(m.getName(), from.extractMemberArray(m));
+      }
+      return to;
+    }
+
+    // from.getSize() <= 0. This usually means that "from" is an ArraySequence, and that we won't know its size until
+    // we iterate over it. extractMemberArray() will do that iteration for us, and then we can use the size of the
```
Comment (DennisHeimbigner) [inline cdm/src/main/java/ucar/ma2/ArrayStructureMA.java]: Typeo in this text

**Comment by DennisHeimbigner** [comment]
These changes are completely opaque to me. So sure go ahead and make the change.

File: cdm/src/main/java/ucar/ma2/ArrayStructureMA.java (hunk 1 above)
Comment (cwardgar) [reply cdm/src/main/java/ucar/ma2/ArrayStructureMA.java]: Really? I don't see it.

**Comment by cwardgar** [comment]
You're the code owner of the DAP test resources. When I tweaked the NcDumpW output, I had to update many of those resources. I assume those edits are okay.

**Pull request merged**